lib/membrane/rtcp/feedback_packet/fir.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Membrane.RTCP.FeedbackPacket.FIR do
+  @moduledoc """
+  Encodes and decodes [Full Intra Request](https://datatracker.ietf.org/doc/html/rfc5104#section-4.3.1) packets.
+  """
+  @behaviour Membrane.RTCP.FeedbackPacket
+
+  @enforce_keys [:target_ssrc, :seq_num]
+  defstruct @enforce_keys
+
+  @impl true
+  def encode(%__MODULE__{} = packet) do
+    <<packet.target_ssrc::32, packet.seq_num, 0::24>>
+  end
+
+  @impl true
+  def encode(packets) when is_list(packets) do
+    Enum.map_join(packets, &encode/1)
+  end
+
+  @impl true
+  def decode(binary) do
+    do_decode(binary, [])
+  end
+
+  defp do_decode(<<>>, acc) do
+    {:ok, Enum.reverse(acc)}
+  end
+
+  defp do_decode(<<target_ssrc::32, seq_num, 0::24, rest::binary>>, acc) do
+    do_decode(rest, [%__MODULE__{target_ssrc: target_ssrc, seq_num: seq_num} | acc])
+  end
+
+  defp do_decode(_binary, _acc) do
+    {:error, :malformed_packet}
+  end
```

Comment:
You can use `for` for it
```suggestion
  def decode(binary) when div(bit_size(binary), 32+8+24) != 0 do
    {:error, :malformed_packet}
  end

  def decode(binary) do
    for <<target_ssrc::32, seq_num::8, 0::24 <- binary>> do
      %__MODULE__{target_ssrc: target_ssrc, seq_num: seq_num}
    end
  end
```

Reply:
`div(bit_size(binary), 32+8+24) != 0` looks ugly and is error-prone, as it requires having the sizes in two places. I'd stick to the original version if there's no way to get rid of that

Reply:
1. Ugliness is in the eye of the beholder, I prefer the div version for stating explicitly what is considered a valid packet and what isn't
2. It's as error prone as the first version where you type sizes from RFC by hand and If you want to do this once, create module attribute. It's not like the RFC is going to change in that matter

Reply:
RFC won't change, but the code may be refactored. Keeping sizes in module attributes requires additional boilerplate, while the purpose of that change is to be more concise

Reply:
>RFC won't change, but the code may be refactored

But the sizes won't

> requires additional boilerplate

If you consider `@` and name a boilerplate then you can start naming variables as `1`,  `2` to avoid boilerplate :stuck_out_tongue: 
Here's the said boilerplate: 
```
@fir_packet_size 32+8+24
...
def decode(binary) when div(bit_size(binary), @fir_packet_size) != 0
```
Besides, It's much less boilerplate than creating a recursive parsing function which requires a wrapper function, `Enum.reverse` and list appending instead of returning value

Reply:
> But the sizes won't

But the code that contains the sizes may be.

> @fir_packet_size 32+8+24

This actually solves nothing, because you still have the sizes in two places.

Reply:
This discussion makes no sense, as the `for` version won't fail when `reserved` part is not set to 0